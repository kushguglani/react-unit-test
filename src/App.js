import './App.css';
import React from 'react';

function App() {
  const [counter, setCounter] = React.useState(0);
  const delayCount = () =>
		setTimeout(() => {
			setCounter(counter + 1);
		}, 500);

  const name="Kush Guglani"
  return (
    <section className="myapp">
    <div>
      <h1>Hi Hello react {name} </h1>
    </div>
    <div>
      <h1>Welcome to unit test case </h1>
      <p>we learn snapshot component </p>
    </div>
    <div>
      <button onClick={()=>setCounter(counter+1)} data-testid='button-up' >Increment</button>
      <button onClick={()=>setCounter(counter-1)} data-testid='button-down' >Decrement</button>

      <button onClick={delayCount} data-testid='button-up-inc' >Increment After 5 sec</button>

      <p data-testid="counter">{counter}</p>
    </div>
    </section>
  );
}

export default App;
