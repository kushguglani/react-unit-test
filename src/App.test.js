import { render, screen, cleanup, fireEvent,waitForElement, } from '@testing-library/react';
import App from './App';
afterEach(cleanup);
describe("testing App snapshot and text",()=>{
  it("should match a snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
  it('renders learn react link', () => {
    render(<App />);
    const linkElement = screen.getByText(/Hello react/i);
    expect(linkElement).toBeInTheDocument();
  });
})

describe("testing buttons and content",()=>{
  it("default counter value should be present",()=>{
    const { getByTestId } = render(<App />);
    expect(getByTestId("counter")).toHaveTextContent(0);
  })

  it("should be enabled", () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId("button-up")).not.toHaveAttribute("disabled");
  });
  xit("should be disabled", () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId("button-down")).toBeDisabled();
  });

  it("increments counter", () => {
    const { getByTestId } = render(<App />);
    fireEvent.click(getByTestId("button-up"));
    expect(getByTestId("counter")).toHaveTextContent("1");
  });
  it("decrements counter", () => {
    const { getByTestId } = render(<App />);
    fireEvent.click(getByTestId("button-down"));
    expect(getByTestId("counter")).toHaveTextContent("-1");
  });

  it("increments async counter", () => {
    const { getByTestId } = render(<App />);
    fireEvent.click(getByTestId("button-up"));
    const counter = waitForElement(() => getByText("1"));
    expect(counter).toHaveTextContent("1");
  });

})


