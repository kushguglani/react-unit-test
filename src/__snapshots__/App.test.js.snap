// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App snapshot and text should match a snapshot 1`] = `
<DocumentFragment>
  <section
    class="myapp"
  >
    <div>
      <h1>
        Hi Hello react Kush Guglani 
      </h1>
    </div>
    <div>
      <h1>
        Welcome to unit test case 
      </h1>
      <p>
        we learn snapshot component 
      </p>
    </div>
    <div>
      <button
        data-testid="button-up"
      >
        Increment
      </button>
      <button
        data-testid="button-down"
      >
        Decrement
      </button>
      <button
        data-testid="button-up-inc"
      >
        Increment After 5 sec
      </button>
      <p
        data-testid="counter"
      >
        0
      </p>
    </div>
  </section>
</DocumentFragment>
`;
